src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java

```diff
@@ -22,50 +21,96 @@
  */
 public abstract class CustomViewDialogFragment<V extends View, T> extends DialogFragment implements DataConsumer<T> {
 
-  private final FragmentActivity activity;
-  private final int viewResourceId;
-  private final ViewRenderer<V, T> renderer;
-  private final DataSource<T> dataSource;
-
-  public CustomViewDialogFragment(FragmentActivity activity, int viewResourceId, ViewRenderer<V, T> renderer,
-      DataSource<T> dataSource) {
-    super();
-    this.activity = activity;
-    this.viewResourceId = viewResourceId;
-    this.renderer = renderer;
-    this.dataSource = dataSource;
-  }
-
-  protected final FragmentActivity getFragmentActivity() {
-    return activity;
-  }
-
-  @Override
-  @SuppressWarnings("unchecked")
-  public V getView() {
-    return (V) super.getView();
-  }
+  private DataSource<T> dataSource;
 
   @Override
   public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
     // Inflate the view identified by the resource id as the view for the fragment.
-    return inflater.inflate(viewResourceId, null, false);
+    return inflater.inflate(getLayoutResourceId(), null, false);
   }
 
   @Override
   public void onActivityCreated(Bundle savedInstanceState) {
     super.onActivityCreated(savedInstanceState);
-    // Start loading data here. This is called here since the view will already have been created by now.
-    if (dataSource != null) {
-      getFragmentActivity().getSupportLoaderManager()
-          .initLoader(0, null, new DataLoader<T>(getActivity(), dataSource, this));
-    }
+    dataSource = createDataSource();
+    startLoad(dataSource, getLoaderId(), this);
+  }
+
+  /**
+   * @return Resource Id for the View used to render the Fragment.
+   */
+  protected abstract int getLayoutResourceId();
+
+  /**
+   * @return View Id of the View that is rendered by the Renderer. Return {@code null} when a renderer is not defined.
+   */
+  protected abstract int getRendererViewId();
+
+  protected V getRendererView() {
+    @SuppressWarnings("unchecked")
+    V view = (V) getView().findViewById(getRendererViewId());
+    return view;
+  }
+
+  /**
+   * Creates a data source used to load the data for the fragment.
+   *
+   * @return The DataSource used to load data for the fragment.
+   */
+  protected abstract DataSource<T> createDataSource();
+
+  /**
+   * Returns a unique Id to be used for the loader that loads the data from the data source.
+   * <p/>
+   * The fragment MUST override this method if there are more than one loader in the activity. Returns 0 by default
+   * with the assumption that there is only one loader being used in the activity.
+   *
+   * @return Unique Id for the loader.
+   */
+  protected int getLoaderId() {
+    return 0;
+  }
+
+  /**
+   * Re-initializes the Fragment.
+   */
+  protected void reinitialize() {
```

Comment:
Yes they are. One just reinitialize the same fragment with the same data, while the other is called from the RefreshFragmentAction passing in new set of arguments. I though of combining them, but have left it different for now. I have a feeling we won't need refresh